### ISSUE TITLE
BUG: Calendar Sessions Return 504 Error Generating Paper Service Document

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8442]
**Describe the Bug**
Occasionally when setting calendars, the operation times out back to the user with an error that the operation could not be completed. Investigation reveals that this seems to happen when the process that generates the package of paper service documents takes longer than is allowed by the lambda running that process.

**Business Impact/Reason for Severity**
Failure to provide paper service documentation to the user results in delay of service to the parties and requires manual retrieval of the paper service document(s). Additionally, there is risk that not all paper service NTDs/clinic letters are being generated.

**In which environment did you see this bug?**
PROD, TEST

**Who were you logged in as?**
Petitionsclerk1

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
Setting calendars

**To Reproduce**
1. Create a trial session and mark as "QC Complete" ~100 cases, at least half of which receive paper service.
2. Click "Set Calendar".
3. You may or may not receive the error; we were most easily able to reproduce this on TEST by engaging in other operations at the same time (i.e., editing another trial session's details, cutting another calendar).

**Expected Behavior**
Set Calendar process completes by presenting the user w/ package of paper documents

**Actual Behavior**
User is presented with error saying the process couldn't be completed

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Ready for Bugs(Created 10-4-21)
Definition used: A failure or flaw in the system which produces an incorrect or undesired result that deviates from the expected result or behavior. (Note: Expected results are use cases that have been documented in past user stories as acceptance criteria and test cases, and do not include strange behavior unrelated to use cases.)

The following criteria must be met in order for the development team to begin work on the bug.

The bug must:
 - Be focused on solving a user problem
 - Contain data for all fields in the bug template, so the team can pick it up and begin working immediately

Process: If the unexpected results are new use cases that have been identified, but not yet built, new acceptance criteria and test cases should be captured in a new user story and prioritized by the product owner. 

If the Court is not able to reproduce the bug, add the “Unable to reproduce” tag. This will provide visibility into the type of support that may be needed by the Court. In the event that the Court cannot reproduce the bug, the Court will work with Flexion to communicate what type of troubleshooting help may be needed.

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's `test` environment if prod-like data is required. Otherwise, deployed to any `experimental` environment for review.

**Comment by ttlenard** [comment]
Testing Feedback @codyseibert 

I did two tests where I was logged in on the court laptop with petitionsclerk6. I was also logged in on my personal laptop as petitionsclerk1.  I set a calendar at roughly the same time on each device.  All sessions had over 100 cases on them.  Both times I ran this the paper service progress bar went over 100% on my personal laptop, which is on a chrome browser.  

Chris and I also did a test where he was petitionsclerk 1 and I was petitionsclerk6. We set the calendar for two different sessions at roughly the same time.  My calendar session finished fine, but Chris' stalled out at the generate notices step.  Here is a screen grab of his console:

![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/aef0f05d-c4cf-417a-8d51-a9767c431e59)

I looked in the logs, and I see this error.
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/cc4e9d59-bfb9-4b8c-aa86-b688a7e4a244)
